salt/modules/win_iis.py

```diff
@@ -1774,3 +1774,173 @@ def list_worker_processes(apppool):
         log.warning('No backups found in output: {0}'.format(cmd_ret))
 
     return ret
+
+
+def get_webapp_settings(name, site, settings):
+    '''
+    Get the value of the setting for the IIS web application.
+    .. note::
+        Params are case sensitive.
+    :param str name: The name of the IIS web application.
+    :param str site: The site name contains the web application.
+        Example: Default Web Site
+    :param str settings: A dictionary of the setting names and their values.
+        Available settings: physicalPath, applicationPool, userName, password
+    Returns:
+        dict: A dictionary of the provided settings and their values.
+    .. versionadded:: Nitrogen
+    CLI Example:
+    .. code-block:: bash
+        salt '*' win_iis.get_webapp_settings name='app0' site='Default Web Site'
+            settings="['physicalPath','applicationPool']"
+    '''
+    ret = dict()
+    pscmd = list()
+    availableSettings = ('physicalPath', 'applicationPool', 'userName', 'password')
+
+    if not settings:
+        log.warning('No settings provided')
+        return ret
+
+    pscmd.append(r'$Settings = @{};')
+
+    # Verify setting is ine predefined settings and append relevant query command per setting key
+    for setting in settings:
+        if setting in availableSettings:
+            if setting == "userName" or setting == "password":
+                pscmd.append(" $Property = Get-WebConfigurationProperty -Filter \"system.applicationHost/sites/site[@name='{0}']/application[@path='/{1}']/virtualDirectory[@path='/']\"".format(site, name))
+                pscmd.append(" -Name \"{0}\" -ErrorAction Stop | select Value;".format(setting))
+                pscmd.append(" $Property = $Property | Select-Object -ExpandProperty Value;")
+                pscmd.append(" $Settings['{0}'] = [String] $Property;".format(setting))
+                pscmd.append(' $Property = $Null;')
+
+            if setting == "physicalPath" or setting == "applicationPool":
+                pscmd.append(" $Property = (get-webapplication {0}).{1};".format(name, setting))
+                pscmd.append(" $Settings['{0}'] = [String] $Property;".format(setting))
+                pscmd.append(r' $Property = $Null;')
+        else:
+            availSetStr = ', '.join(availableSettings)
+            message = 'Unexpected setting:' + setting + '. Available settings are: ' + availSetStr
+            raise SaltInvocationError(message)
+
+    pscmd.append(' $Settings')
+    # Run commands and return data as json
+    cmd_ret = _srvmgr(cmd=str().join(pscmd), return_json=True)
+
+    # Update dict var to return data
+    try:
+        items = json.loads(cmd_ret['stdout'], strict=False)
+
+        if isinstance(items, list):
+            ret.update(items[0])
+        else:
+            ret.update(items)
+    except ValueError:
+        log.error('Unable to parse return data as Json.')
+
+    if None in ret.viewvalues():
```

Comment:
We need a `from salt.ext import six` added to the import section near the top of this file and then this line should be changed to `six.viewvalues(ret)`.